### PR TITLE
Migrate AudioService to Jellyfin.Api

### PR DESCRIPTION
**Pull request by Ullmie02**
After trying for over a week to migrate the first streaming endpoint, I am slowly starting to understand how the streaming API works and what everything is intended to do.
But now I have come to a point where I don't know what to do next. 

**Changes I am fairly certain that will work**
- `FileStreamResponseHelpers.cs` has multiple methods to return the media files, depending on what method you want to use (direct stream, stream from remote server, transcoding)
   -  ~Caching for media files doesn't work (this has a TODO in `FileStreamResponseHelpers.cs`)~
- `TranscodingJobHelper.cs` is now a singleton and can be injected via DI. 

**Problems I have created/discoverd while migrating this API service**
- As the request parameters are no longer in their own object, passing some of them around is a bit complex. `StreamState` had a property which could be such a request object. But as such a object does not exist in Jellyfin.Api, some request parameters can't get to the place where they are needed.
- The audio endpoint has parameters like `framerate`, `width`, `height`, `subtitleStreamIndex`, etc. which make absolutely no sense for an audio file. I'm not sure if they can be safely removed, so I left them in.
- The old endpoint has a parameter `params` which gets parsed and some of the request parameters get assigned again? I don't understand why this is used or if this is still used. (see StreamingHelpers.cs L585)
- I'm not even sure if it will work after we fix all the compiler errors mentioned above.

I know this PR is huge and I am very sorry, but if we want to migrate the playback API to Jellyfin.Api, someone has to start with the first service which probably lays the groundwork for the rest of them.
This needs more discussion on how to solve those problems and if my architectural decision about making TranscodingJobHelper a singleton was right.
I am open about feedback and improvements, as I don't know how to solve those problems.

### CODE REVIEW

**Comment by Ullmie02** [comment]
Caching isn't used so I removed it.
The Content length header gets added by the framework automatically.

File: Jellyfin.Api/Helpers/FileStreamResponseHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        /// <param name="state">The current <see cref="StreamState"/>.</param>
+        /// <param name="isHeadRequest">Whether the current request is a HTTP HEAD request so only the headers get returned.</param>
+        /// <param name="controller">The <see cref="ControllerBase"/> managing the response.</param>
+        /// <returns>A <see cref="Task{ActionResult}"/> containing the API response.</returns>
+        public static async Task<ActionResult> GetStaticRemoteStreamResult(
+            StreamState state,
+            bool isHeadRequest,
+            ControllerBase controller)
+        {
+            HttpClient httpClient = new HttpClient();
```
Comment (crobibero) [inline Jellyfin.Api/Helpers/FileStreamResponseHelpers.cs:32]: Can't initialize a new HttpClient this way- https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/

Comment (cvium) [reply Jellyfin.Api/Helpers/FileStreamResponseHelpers.cs:32]: That article is outdated. The issue it highlights is still valid of course, but the solution should be ignored. Use this https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-3.1#basic-usage

**Comment by dkanada** [comment]
@Ullmie02 does a parameter currently exist to force direct play or would it be easy to add such an option during the migration? Saw the comments above and it sounded like a possibility, would be super nice for audio.

**Comment by Ullmie02** [comment]
@dkanada if you supply `static=true` as a query parameter, you should get the file without any transcoding or re-encoding

File: Jellyfin.Api/Models/StreamingDtos/StreamState.cs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+                if (EncodingHelper.IsCopyCodec(OutputVideoCodec))
+                {
+                    var userAgent = UserAgent ?? string.Empty;
+
+                    if (userAgent.IndexOf("AppleTV", StringComparison.OrdinalIgnoreCase) != -1 ||
+                        userAgent.IndexOf("cfnetwork", StringComparison.OrdinalIgnoreCase) != -1 ||
+                        userAgent.IndexOf("ipad", StringComparison.OrdinalIgnoreCase) != -1 ||
+                        userAgent.IndexOf("iphone", StringComparison.OrdinalIgnoreCase) != -1 ||
+                        userAgent.IndexOf("ipod", StringComparison.OrdinalIgnoreCase) != -1)
```
Comment (crobibero) [inline Jellyfin.Api/Models/StreamingDtos/StreamState.cs:95]:
```suggestion
                    if (userAgent.IndexOf("AppleTV", StringComparison.OrdinalIgnoreCase) != -1
                        || userAgent.IndexOf("cfnetwork", StringComparison.OrdinalIgnoreCase) != -1
                        || userAgent.IndexOf("ipad", StringComparison.OrdinalIgnoreCase) != -1
                        || userAgent.IndexOf("iphone", StringComparison.OrdinalIgnoreCase) != -1
                        || userAgent.IndexOf("ipod", StringComparison.OrdinalIgnoreCase) != -1)
```

File: Jellyfin.Api/Controllers/AudioController.cs [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+                            }
+                    }).ConfigureAwait(false);
+                }
+                finally
+                {
+                    state.Dispose();
+                }
+            }
+            */
+
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/AudioController.cs:375]:
```suggestion
```

File: Jellyfin.Api/Controllers/AudioController.cs [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+                VideoCodec = videoCodec,
+                SubtitleCodec = subtitleCodec,
+                TranscodeReasons = transcodingReasons,
+                AudioStreamIndex = audioStreamIndex,
+                VideoStreamIndex = videoStreamIndex,
+                Context = context,
+                StreamOptions = streamOptions
+            };
+
+            var state = await StreamingHelpers.GetStreamingState(
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/AudioController.cs:263]:
```suggestion
            using var state = await StreamingHelpers.GetStreamingState(
```

And can get rid of the `using(state)`, and manual dispose

**Pull request merged**